### PR TITLE
Hw fruit shop tests

### PR DESCRIPTION
**Pull request by MarkGodnyi**

### CODE REVIEW

File: src/test/java/core/basesyntax/service/CsvFileReaderServiceImplTest.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+                + expected + " but was: "
+                + actual, expected, actual);
+    }
+
+    @Test(expected = RuntimeException.class)
+    public void readFromFile_fileDoesNotExist_notOk() {
+        readerService.readFromFile(PATH_TO_FILE);
+    }
+
+    public static void writeToFileText() {
```
Comment (MaliukDaria) [inline src/test/java/core/basesyntax/service/CsvFileReaderServiceImplTest.java:44]:
```suggestion
    private static void writeTestDataToFile() {
```

File: src/test/java/core/basesyntax/service/CsvFileReaderServiceImplTest.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        readerService = new CsvFileReaderServiceImpl();
+    }
+
+    @After
+    public void tearDown() {
+        new File(PATH_TO_FILE).delete();
+    }
+
+    @Test
+    public void readFromFile_defaultCase_Ok() {
```
Comment (MaliukDaria) [inline src/test/java/core/basesyntax/service/CsvFileReaderServiceImplTest.java:30]:
```suggestion
    public void readFromFile_validData_Ok/readFromFile_Ok() {
```

File: src/test/java/core/basesyntax/service/CsvFileReaderServiceImplTest.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+                + actual, expected, actual);
+    }
+
+    @Test(expected = RuntimeException.class)
+    public void readFromFile_fileDoesNotExist_notOk() {
+        readerService.readFromFile(PATH_TO_FILE);
+    }
+
+    public static void writeToFileText() {
+        Util.createNewFolder();
```
Comment (MaliukDaria) [inline src/test/java/core/basesyntax/service/CsvFileReaderServiceImplTest.java:45]: Lets move this `Util.createNewFolder();` to `readFromFile_defaultCase_Ok()` method

File: src/test/java/core/basesyntax/service/CsvFileWriterServiceImplTest.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    public void writeToFile_defaultCase_Ok() {
+        Util.createNewFolder();
+        writerService.writeToFile(OUTPUT_FILE_PATH, Util.createTextFromLines(Util.REPORT));
+        List<String> lines;
+        try (BufferedReader reader = new BufferedReader(new FileReader(OUTPUT_FILE_PATH))) {
+            lines = reader.lines().collect(Collectors.toList());
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read from file: " + OUTPUT_FILE_PATH, e);
+        }
+        String expected = Util.createTextFromLines(Util.REPORT);
```
Comment (MaliukDaria) [inline src/test/java/core/basesyntax/service/CsvFileWriterServiceImplTest.java:40]: could we call `Util.createTextFromLines(Util.REPORT)` method only ones here?

File: src/test/java/core/basesyntax/service/CsvFileWriterServiceImplTest.java [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+        String actual = lines.stream().collect(Collectors.joining(System.lineSeparator()));
+        assertEquals("Expected should be equal to "
+                + expected + " but was: "
+                + actual, expected, actual);
```
Comment (MaliukDaria) [inline src/test/java/core/basesyntax/service/CsvFileWriterServiceImplTest.java:44]: lets make exception message more informative? (It is the same in all test classes and tests.... it should briefly describe what are we testing here)

File: src/test/java/core/basesyntax/service/CsvFileWriterServiceImplTest.java [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+    }
+
+    @Test(expected = RuntimeException.class)
+    public void writeToFile_FileCanNotBeReached_notOk() {
+        Util.createNewFolder();
```
Comment (MaliukDaria) [inline src/test/java/core/basesyntax/service/CsvFileWriterServiceImplTest.java:49]: do we need to create this folder for this test?

File: src/test/java/core/basesyntax/service/FruitTransactionsServiceImplTest.java [hunk 7]
```diff
@@ -0,0 +1,39 @@
+package core.basesyntax.service;
+
+import static org.junit.Assert.assertEquals;
+
+import java.util.ArrayList;
+import java.util.List;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class FruitTransactionsServiceImplTest {
```
Comment (MaliukDaria) [inline src/test/java/core/basesyntax/service/FruitTransactionsServiceImplTest.java:10]: lets also add tests for empty file

File: src/test/java/core/basesyntax/service/ReportCreatorServiceImplTest.java [hunk 8]
```diff
@@ -0,0 +1,34 @@
+package core.basesyntax.service;
+
+import static org.junit.Assert.assertEquals;
+
+import core.basesyntax.db.FruitStorage;
+import org.junit.After;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class ReportCreatorServiceImplTest {
```
Comment (MaliukDaria) [inline src/test/java/core/basesyntax/service/ReportCreatorServiceImplTest.java:10]: lets add test for empty storage

File: src/test/java/core/basesyntax/strategy/OperationStrategyImplTest.java [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+    private static Map<FruitTransaction.Operation, OperationHandler> operationHandlerMap;
+
+    @BeforeClass
+    public static void beforeClass() {
+        initializeMap();
+        strategy = new OperationStrategyImpl(operationHandlerMap);
+    }
+
+    @Test
+    public void get_differentOperationHandlers_Ok() {
```
Comment (MaliukDaria) [inline src/test/java/core/basesyntax/strategy/OperationStrategyImplTest.java:27]: I think it's betetr to divide this logic to separate test methods

File: src/test/java/core/basesyntax/strategy/operation/OperationHandlersTest.java [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+        initializeFruitTransactions();
+    }
+
+    @After
+    public void tearDown() {
+        FruitStorage.fruitsMap.clear();
+    }
+
+    @Test
+    public void handle_defaultCase_Ok() {
```
Comment (MaliukDaria) [inline src/test/java/core/basesyntax/strategy/operation/OperationHandlersTest.java:33]: I think It's better to add tests for all operation handlers separately

**Review by MaliukDaria — CHANGES_REQUESTED** [review]

File: src/test/java/core/basesyntax/service/CsvFileReaderServiceImplTest.java [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+                + expected + " but was: "
+                + actual, expected, actual);
+    }
+
+    @Test(expected = RuntimeException.class)
+    public void readFromFile_fileDoesNotExist_notOk() {
+        readerService.readFromFile(PATH_TO_FILE);
+    }
+
+    public static void writeTestDataToFile() {
```
Comment (MaliukDaria) [inline src/test/java/core/basesyntax/service/CsvFileReaderServiceImplTest.java:45]:
```suggestion
    private void writeTestDataToFile() {
```

File: src/test/java/core/basesyntax/service/FruitTransactionsServiceImplTest.java [hunk 12]
```diff
@@ -0,0 +1,50 @@
...
+    private static FruitTransactionsService fruitTransactionsService;
+    private static List<FruitTransaction> fruitTransactions;
+    private static List<String> emptyList;
+
+    @BeforeClass
+    public static void beforeClass() {
+        fruitTransactionsService = new FruitTransactionsServiceImpl();
+        initializeFruitTransactions();
+        emptyList = new ArrayList<>();
+    }
```
Comment (MaliukDaria) [inline src/test/java/core/basesyntax/service/FruitTransactionsServiceImplTest.java:20]:
```suggestion
    private static List<String> emptyList;
    private List<FruitTransaction> fruitTransactions;
    
    @BeforeClass
    public static void beforeClass() {
        fruitTransactionsService = new FruitTransactionsServiceImpl();
        emptyList = new ArrayList<>();
    }
    
    @Before
    public void init() {
        initializeFruitTransactions();
    }
```

It is better to initialize fruitTransactions before every test, so as not to transfer the state of fruitTransactions list from one test to another (same in other such places)

File: src/test/java/core/basesyntax/service/ReportCreatorServiceImplTest.java [hunk 13]
```diff
@@ -0,0 +1,43 @@
...
+    private static ReportCreatorService reportCreatorService;
+
+    @BeforeClass
+    public static void beforeClass() {
+        reportCreatorService = new ReportCreatorServiceImpl();
+    }
+
+    @Test
+    public void makeReport_defaultCase_Ok() {
```
Comment (MaliukDaria) [inline src/test/java/core/basesyntax/service/ReportCreatorServiceImplTest.java:19]:
```suggestion
    public void makeReport_validData_Ok/makeReport_Ok() {
```

File: src/test/java/core/basesyntax/strategy/operation/OperationHandlersTest.java [hunk 14]
```diff
@@ -0,0 +1,94 @@
...
+                        + expectedBananas + " but was: "
+                        + actualBananas,
+                expectedBananas,
+                actualBananas);
+    }
+
+    @Test
+    public void handle_purchaseOperationHandler_Ok() {
+        PURCHASE_OPERATION_HANDLER.handle(purchaseFruitTransaction);
+        int expectedBananas = -20;
```
Comment (MaliukDaria) [inline src/test/java/core/basesyntax/strategy/operation/OperationHandlersTest.java:48]: It is strange to expect negative quantity of fruits, you can put some fruits to storage directly first, an then get them through handler

**Review by MaliukDaria — CHANGES_REQUESTED** [review]

**Review by aaananas — APPROVED** [review]
